### ISSUE TITLE
Checkstyle fails when using Lombok's @FieldDefaults

### ISSUE THREAD

**Issue by ErrorsAndGlitches** [checkstyle/checkstyle#5197]
**/var/tmp $ cat YOUR_FILE.java**
```
@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
public class Foo {
  String secretString;
}
```

**/var/tmp $ cat config.xml**
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name = "Checker">
  <property name="severity" value="error"/>

  <module name="TreeWalker">
    <module name="JavadocVariable">
      <property name="scope" value="package"/>
    </module>
  </module>

</module>
```

**/var/tmp $ java -Duser.language=en -Duser.country=US -jar ./checkstyle-8.3-all.jar foo.java -c test.xml**
```
Starting audit...
[ERROR] ...../foo.java:5:3: Missing a Javadoc comment. [JavadocVariable]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

**Expected Behavior**: No errors should occur.

--------------

**Comment by romani** [comment]
http://checkstyle.sourceforge.net/config_javadoc.html#JavadocVariable
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocVariableCheck.java - this javadoc check does not use regexp for parsing, so it is not deprecated.

from documentation:
> `scope | visibility scope where Javadoc comments are checked`

so behavior is expected, as field (with package private scope) does not have javadoc comment - he is violation .

================

The only problem I see is unsync of Xdoc(html documentation) and code for `tokens`
It is now:
`tokens | tokens to check | subset of tokens               ENUM_CONSTANT_DEF. | ENUM_CONSTANT_DEF. | 3.0 `

tokens should be VARIABLE_DEF and ENUM_CONSTANT_DEF .

**Comment by romani** [comment]
@rnveach ,

```
$ git diff
diff --git a/src/xdocs/config_javadoc.xml b/src/xdocs/config_javadoc.xml
index c1ade7f..517b3d5 100644
--- a/src/xdocs/config_javadoc.xml
+++ b/src/xdocs/config_javadoc.xml
@@ -1309,9 +1309,11 @@ public boolean isSomething()
             <td>tokens</td>
             <td>tokens to check</td>
             <td>subset of tokens
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#VARIABLE_DEF">VARIABLE_DEF</a>,
               <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_CONSTANT_DEF">ENUM_CONSTANT_DEF</a>.
             </td>
             <td>
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#VARIABLE_DEF">VARIABLE_DEF</a>,
               <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_CONSTANT_DEF">ENUM_CONSTANT_DEF</a>.
             </td>
             <td>3.0</td>
```

result in:
`[ERROR]   XdocsPagesTest.testAllCheckSections:287->validateCheckSection:363->validatePropertySection:439->validatePropertySectionProperties:546->validatePropertySectionPropertyTokens:614 config_javadoc.xml section 'JavadocVariable' should have all the acceptable tokens expected:<subset of tokens []ENUM_CONSTANT_DEF.> but was:<subset of tokens [VARIABLE_DEF, ]ENUM_CONSTANT_DEF.>
`

looks like where is problem in Xdoc validation.

**Comment by rnveach** [comment]
@romani `VARIABLE_DEF` is a required token for `JavadocVariable`. It is not optional.
XDoc only lists optional tokens.
Test is correct.

**Comment by romani** [comment]
@rnveach , you are right.

I am closing the issue
